soft4pes/control/lin/lin_curr_ctr.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+from soft4pes.utils.conversions import alpha_beta_2_dq, dq_2_abc
+
+
+class RLGridPICurrCtr:
+    """
+    PI current controller for grid-connected converter with RL load.
+    
+    Attributes
+    ----------
+    Rg : float
+        Resistance [p.u.].
+    Xg : float
+        Reactance [p.u.].       
+    base : base-value object
+        Base values.
+    Ts : float
+        Sampling interval [s].
+    Ts_pu : float
+        Sampling interval [p.u.].
+    alpha_c : float
+        Controller bandwidth [p.u.]
+    k_p : float
+        Proportional gain [p.u.]
+    k_i : float
+        Proportional gain [p.u.]
+    integral_error_d : float
+        Current error instance in d-frame [p.u.].
+    integral_error_q : float
+        Current error instance in q-frame [p.u.].                               
+    i_ref_seq_dq : Sequence
+        Current reference sequence instance in dq-frame [p.u.].   
+    """
+
+    def __init__(self, sys, base, Ts, i_ref_seq_dq):
+        """
+        Initialize a PICurrCtr.
+
+        Parameters
+        ----------
+        sys : system object
+           System model.
+        base : base-value object
+           Base values.
+        Ts : float
+           Sampling interval [s].
+        i_ref_seq_dq : Sequence
+           Current reference sequence instance in dq-frame [p.u.].
+        """
+
+        self.Xg = sys.Xg
+        self.Rg = sys.Rg
+        self.Ts = Ts
+        self.Ts_pu = self.Ts * base.w
+        self.alpha_c = 2 * np.pi / 10 / self.Ts_pu  # Controller bandwidth (10x crossover frequency)
+        self.k_p = self.alpha_c * self.Xg  # Proportional gain
+        self.k_i = self.alpha_c * self.Rg  # Integral gain
+        self.integral_error_d = 0
+        self.integral_error_q = 0
+        self.i_ref_seq_dq = i_ref_seq_dq
+
+    def __call__(self, sys, conv, t):
+        """
+        Perform control.
+
+        Parameters
+        ----------
+        sys : system object
+            System model.
+        conv : converter object
+            Converter model.
+        t : float
+            Current time [s].
+
+        Returns
+        -------
+        1 x 3 ndarray of floats
+            Modulating signal.
+        """
+
+        # Get the grid voltage
+        vg = sys.get_grid_voltage(t)
```

Comment:
Not used anywhere

Reply:
the vg is used to obtain theta to convert alpha beta to dq and other conversions. Do you have another suggestion?

Reply:
My mistake, this is ok as it is